### PR TITLE
Add OpenFOAM language server

### PR DESCRIPTION
**Pull request by FoamScience**
Adding [foam-language-server](https://github.com/FoamScience/foam-language-server) to be seamlessly installed and configured with lsp-installer

### CODE REVIEW

File: lua/nvim-lsp-installer/servers/foam/init.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+return function(name, root_dir)
+	return server.Server:new({
+		name = name,
+		root_dir = root_dir,
+		homepage = "https://github.com/FoamScience/foam-language-server",
+		languages = { "foam", "OpenFOAM" },
+		installer = npm.packages({ "foam-language-server" }),
+		default_options = {
+			--cmd_env = npm.env(root_dir),
+			cmd = { path.concat({ root_dir, "node_modules", "foam-language-server", "bin", "foam-ls" }), "--stdio" },
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/foam/init.lua:14]: Whenever it's possible, we don't override the `cmd` but instead rely on the defaults provided by lspconfig. All we have to do is ensure that the executable exists in path, which with npm packages is done like so:

```suggestion
			cmd_env = npm.env(root_dir),
```

**Review by williamboman — COMMENTED** [review]
Cool! We'll have to wait for the lspconfig PR (https://github.com/neovim/nvim-lspconfig/pull/1643) to land first!

The code style is enforced by [`stylua`](https://github.com/JohnnyMorganz/StyLua) btw! If you change the source branch to something else than the `main` branch of your fork, I could apply the code style for you in a commit in case you don't have stylua installed

File: lua/nvim-lsp-installer/servers/foam/init.lua (hunk 1 above)
Comment (FoamScience) [reply lua/nvim-lsp-installer/servers/foam/init.lua]: OK, this is not working for me;
The language server is downloaded to `~/.local/share/nvim/lsp_servers/foam` with its executable.
But without overriding the cmd, I get 0 clients attached to buffer. Any idea what's going on, I'm using configuration from (neovim/nvim-lspconfig#1643)
Even with globally installed package, if I don't override the cmd, the server is not getting fired up; a little confused I must say.

Comment (williamboman) [reply lua/nvim-lsp-installer/servers/foam/init.lua]: I believe this happens if the server is unable to initialize

Comment (FoamScience) [reply lua/nvim-lsp-installer/servers/foam/init.lua]: OFC I forgot `--stdio`; thanks; all good now

File: lua/nvim-lsp-installer/servers/foam_ls/init.lua [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+return function(name, root_dir)
+    return server.Server:new {
+        name = name,
+        root_dir = root_dir,
+        homepage = "https://github.com/FoamScience/foam-language-server",
+        languages = { "foam", "OpenFOAM" },
+        installer = npm.packages { "foam-language-server" },
+        default_options = {
+            cmd = { path.concat { root_dir, "node_modules", "foam-language-server", "bin", "foam-ls" }, "--stdio" },
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/foam_ls/init.lua:13]: By providing the PATH to the executable, we can rely on the default cmd in `lspconfig` instead. With npm installations, there's a shorthand like this:

```suggestion
            cmd_env = npm.env(root_dir),
```

**Review by williamboman — APPROVED** [review]
Just one final thing, otherwise LGTM! I hope CI is happy by now

File: lua/nvim-lsp-installer/servers/foam_ls/init.lua (hunk 2 above)
Comment (FoamScience) [reply lua/nvim-lsp-installer/servers/foam_ls/init.lua]: I swear I did this yesterday lol but oh well!

**Review by williamboman — APPROVED** [review]

File: lua/nvim-lsp-installer/servers/foam_ls/init.lua (hunk 2 above)
Comment (williamboman) [reply lua/nvim-lsp-installer/servers/foam_ls/init.lua]: git never lies 😛 !

**Pull request merged**